### PR TITLE
[DONT MERGE] Notifying directly from io threads

### PR DESCRIPTION
**Pull request by pveentjer**

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/SerializationService.java [hunk 1]
```diff
@@ -42,6 +42,8 @@
 
     <T> T toObject(Object data);
 
+    <T> T bytesToObject(byte[] bytes, int dataOffset);
```
Comment (toktarev) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/SerializationService.java]: Can we add also int length here ?
or separate method
+    <T> T bytesToObject(byte[] bytes, int dataOffset, int length);
?

Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/SerializationService.java]: What would be the purpose of the length? The deserialization step determines itself how much data it is going to read.

Comment (toktarev) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/SerializationService.java]: So we always rely on the fact that object is totally presented in byte[] array ?

What is the goal of dataOffset in this case ?

Comment (toktarev) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/SerializationService.java]: I see, it is correct

**Comment by pveentjer** [comment]
run-lab-run

**Comment by pveentjer** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/SerializationService.java (hunk 1 above)
Comment (toktarev) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/SerializationService.java]: I am really appreciative for this method
I was going to implement it these days :) Thanks

**Pull request closed without merging**